api/server/router/container/container_routes.go

```diff
@@ -920,6 +927,43 @@ func epConfigForNetMode(
 	return ep, nil
 }
 
+// validatePortBindings checks that all PortBindings are valid, and have
+// a protocol specified.
+//
+// Up until v28.3, the Engine would accept port bindings with no protocol
+// specified. In that case, it would use TCP as the default protocol. This
+// wasn't specified in the API spec, leading to interoperability issues.
+//
+// Starting with Engine v28.3, a warning is returned when the protocol is
+// missing for all API versions up until v1.52. Starting with that API version,
+// it's a hard error.
+func validatePortBindings(hostConfig *container.HostConfig, version string) ([]string, error) {
+	if len(hostConfig.PortBindings) == 0 {
+		return nil, nil
+	}
+
+	var errs []error
+	for portSpec := range hostConfig.PortBindings {
+		parts := strings.Split(string(portSpec), "/")
```

Comment:
Can we use `strings.Cut` and check if the specified scheme is one we support? It's effectively what we do in https://github.com/docker/go-connections/blob/b071e04318669a6f5eeff43d4d85659f9d8cbb2a/nat/nat.go#L100-L110